### PR TITLE
misc: first pipelines for concourse.

### PR DESCRIPTION
**Pull request by grignards**
#4828

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done => not seen as necessary

### CODE REVIEW

File: concourse/scripts/docker-lib.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+SKIP_PRIVILEGED=${SKIP_PRIVILEGED:-false}
+STARTUP_TIMEOUT=${STARTUP_TIMEOUT:-20}
+DOCKER_DATA_ROOT=${DOCKER_DATA_ROOT:-/scratch/docker}
+
+sanitize_cgroups() {
+  mkdir -p /sys/fs/cgroup
+  mountpoint -q /sys/fs/cgroup || \
+    mount -t tmpfs -o uid=0,gid=0,mode=0755 cgroup /sys/fs/cgroup
+
+  mount -o remount,rw /sys/fs/cgroup
```
[the remark this answers is not in the document]
Comment (grignards) [reply concourse/scripts/docker-lib.sh:14]: Added as a task in the meta-issue for improving this first shot: https://github.com/tracim/tracim/issues/4886

File: concourse/scripts/backend-lint [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+source "$HOME/.bashrc"
+python_version=${1:-3.6.14}
+pyenv shell "$python_version"
+
+cd backend
+pip install -r requirements-devtool.txt
+flake8 --version
```
[the remark this answers is not in the document]
Comment (grignards) [reply concourse/scripts/backend-lint:10]: Added as a task in #4886.

File: concourse/scripts/install_backend_packages [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# !/bin/bash
+set -e
+
+apt-get update
+apt-get install -qy \
```
[the remark this answers is not in the document]
Comment (grignards) [reply concourse/scripts/install_backend_packages:5]: I created an issue about this debt: https://github.com/tracim/tracim/issues/4887

File: concourse/scripts/docker-lib.sh [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  if [ "$SKIP_PRIVILEGED" = "false" ]; then
+    sanitize_cgroups
+
+    # check for /proc/sys being mounted readonly, as systemd does
+    if grep '/proc/sys\s\+\w\+\s\+ro,' /proc/mounts >/dev/null; then
+      mount -o remount,rw /proc/sys
+    fi
+  fi
+
+  local mtu=$(cat /sys/class/net/$(ip route get 8.8.8.8|awk '{ print $5 }')/mtu)
```
[the remark this answers is not in the document]
Comment (grignards) [reply concourse/scripts/docker-lib.sh]: Not a hard one as the address is only used to get the interface which would be used to route packets to this address.
Nonetheless I added it as a variable.

**Review by raphj — APPROVED** [review]

**Review by inkhey — APPROVED** [review]

File: concourse/scripts/backend-full-tests [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+
+python_version=${1:-3.6.14}
+database=${2:-all}
+
+source "$HOME/.bashrc"
+pyenv shell "$python_version"
```
[the remark this answers is not in the document]
Comment (grignards) [reply concourse/scripts/backend-full-tests:8]: Wrote an issue: #4889

**Pull request merged**